src/client/Microsoft.Identity.Client/OAuth2/DeviceAuthJWTResponse.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Microsoft.Identity.Client.Utils;
+using Microsoft.Identity.Json;
+
+namespace Microsoft.Identity.Client.OAuth2
+{
+    [JsonObject]
+    internal class DeviceAuthHeader
+    {
+        public DeviceAuthHeader(string base64EncodedCertificate)
+        {
+            this.Alg = "RS256";
+            this.Type = "JWT";
+            this.X5c = new List<string>();
+            this.X5c.Add(base64EncodedCertificate);
+        }
+
+        [JsonProperty("x5c")]
+        public IList<string> X5c { get; set; }
+
+        [JsonProperty("typ")]
+        public string Type { get; set; }
+
+        [JsonProperty("alg")]
+        public string Alg { get; private set; }
+    }
+
+    [JsonObject]
+    internal class DeviceAuthPayload
+    {
+        public DeviceAuthPayload(string audience, string nonce)
+        {
+            this.Nonce = nonce;
+            this.Audience = audience;
+            var timeSpan = (DateTime.UtcNow - new DateTime(1970, 1, 1, 0, 0, 0));
```

Comment:
>new DateTime(1970, 1, 1, 0, 0, 0) [](start = 46, length = 33)

would it make sense to have this as a named lazy var (or perhaps even static)?

Reply:
yes it does, ill change